io/genbank/genbank.go

```diff
@@ -33,24 +34,143 @@ GBK specific IO related things begin here.
+	Sequence string // will be changed and include reader, writer, and byte slice.
+}
+
+type Meta struct {
+	Date                 string            `json:"date"`
+	Definition           string            `json:"definition"`
+	Accession            string            `json:"accession"`
+	Version              string            `json:"version"`
+	Keywords             string            `json:"keywords"`
+	Organism             string            `json:"organism"`
+	Source               string            `json:"source"`
+	Origin               string            `json:"origin"`
+	Locus                Locus             `json:"locus"`
+	References           []Reference       `json:"references"`
+	Other                map[string]string `json:"other"`
+	Name                 string            `json:"name"`
+	SequenceHash         string            `json:"sequence_hash"`
+	SequenceHashFunction string            `json:"hash_function"`
+	CheckSum             [32]byte          `json:"checkSum"` // blake3 checksum of the parsed file itself. Useful for if you want to check if incoming genbank/gff files are different.
+}
+
+type Feature struct {
+	Type                 string            `json:"type"`
+	Description          string            `json:"description"`
+	Attributes           map[string]string `json:"attributes"`
+	SequenceHash         string            `json:"sequence_hash"`
+	SequenceHashFunction string            `json:"hash_function"`
+	Sequence             string            `json:"sequence"`
+	GbkLocationString    string            `json:"gbk_location_string"`
+	Location             poly.Location     `json:"sequence_location"`
+	ParentSequence       *Genbank          `json:"-"`
+}
+
+// Reference holds information one reference in a Meta struct.
+type Reference struct {
+	Index   string `json:"index"`
+	Authors string `json:"authors"`
+	Title   string `json:"title"`
+	Journal string `json:"journal"`
+	PubMed  string `json:"pub_med"`
+	Remark  string `json:"remark"`
+	Range   string `json:"range"`
+}
+
+// Locus holds Locus information in a Meta struct.
+type Locus struct {
+	Name             string `json:"name"`
+	SequenceLength   string `json:"sequence_length"`
+	MoleculeType     string `json:"molecule_type"`
+	GenbankDivision  string `json:"genbank_division"`
+	ModificationDate string `json:"modification_date"`
+	SequenceCoding   string `json:"sequence_coding"`
+	Circular         bool   `json:"circular"`
+	Linear           bool   `json:"linear"`
+}
+
+func (genbank Genbank) GetMeta() (Meta, error) {
+	return genbank.Meta, nil
+}
+
+func (genbank Genbank) GetFeatures() ([]Feature, error) {
+	return genbank.Features, nil
+}
+
+func (genbank *Genbank) GetSequence() (string, error) {
+	return genbank.Sequence, nil
+}
+
+func (sequence *Genbank) AddFeature(feature *Feature) error {
+	feature.ParentSequence = sequence
+	sequence.Features = append(sequence.Features, *feature)
+	return nil
+}
+
+func (feature Feature) GetSequence() (string, error) {
```

Comment:
Even though this and the method below should satisfy the Feature interface build process throws this error implying that it doesn't. 
![Screen Shot 2021-12-08 at 3 35 57 PM](https://user-images.githubusercontent.com/7042260/145457430-7bba326c-f64f-408f-9e9c-791250a39e18.png)

Reply:
@TimothyStiles does it make sense to have multiple classes/interfaces with the same name `Feature`. I think the names should be more descriptive

Reply:
@rkrishnasanka in this case I'd argue yes just because each `Feature` struct is within its own subpackage though I am wondering if namespace collision is the issue here.

I did try naming them more specifically, i.e `GenbankFeature` but that didn't alleviate the issue and unfortunately this convention would create a stutter from the user's end so they'd have to create a new one with `genbank.GenbankFeature` instead of just `genbank.Feature`

The `InvalidIfaceAssign` error is supposed to mean that the struct being referenced isn't correctly implementing the interface needed but I can't seem to figure out why it's wrong.

Reply:
Yes while it doesn't directly address your issue here I think it's a better code practice to have more descriptive names. I can imagine scenarios where one might need to load both the Feature classes into same file and have to keep fluttering back and forth to ensure things are being used correctly.